---
title: "SciPy 2020"
params:
  subtitle: "Paper discussion"
  authorA: "Luna L. Sanchez Reyes"
  authorB: "Martha Kandziora"
  authorC: "Emily Jane McTavish"
  institute: "University of California, Merced"
  place: "The Internet"
  date: "July 6-12, 2020"
  event: "SciPy2020 Conference"
output:
  xaringan::moon_reader:
    lib_dir: libs
    css: ["kunoichi", "ninjutsu", "metropolis-fonts", "ninjutsu-luna-extras.css"]
    seal: false
    self_contained: false
    nature:
      ratio: "16:9"
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

class: left, middle, blue-gray

<img align="left" width="250" src="https://cdn.rawgit.com/snacktavish/physcraper/master/docs/physcraper.svg">


```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
knitr::opts_chunk$set(echo = FALSE)
options(knitr.duplicate.label = 'allow')
```

.font3[**Physcraper**]

.font2[Continual .red[updating] of evolutionary estimates in the Open Tree of Life]

<br/>
### .red[<sup>*</sup>] `r params$authorA` .red[&#183;]  `r params$authorB` .red[&#183;] `r params$authorC`
<br/>
### `r params$institute`
<br/>


`r params$event` .red[&#183;] `r params$date` .red[&#183;] `r params$place`

Slides last updated on  `r format(Sys.Date(), format = "%B %d, %Y")` .red[&#183;] Find the code for them in [`r anicon::faa("github", animate="passing")`  here]()

.red[*] Speaker contact [@LunaSare](https://www.lunasare.com/) on [Github `r anicon::faa("github", animate="float")`](https://github.com/LunaSare) and [Twitter `r anicon::faa("twitter", animate="float")`](https://twitter.com/LunaSare)
---

background-image: url("bg/2-past-future1.png")
background-position: center
background-size: contain

???

# To understand our future you need to understand where you come from.

---
background-image: url("bg/3-diversity1.png")
background-position: center
background-size: contain

???

# As biologists, we want to understand how all living things are connected evolutionarily

---
background-image: url("bg/4-diversity2.png")
background-position: center
background-size: contain

class: split-20, center

.row[.content[
# Let's talk scale... how many organisms are there?

]]
.row.split-two[
.column[.content.center[
<br/><br/><br/>
### &emsp; &emsp; &emsp;  Estimates range between 
## &emsp; &emsp; &emsp; .orange[3 million] .font_small.gray[(Mayr 2010)]
### &emsp; &emsp; &emsp;  and 
## &emsp; &emsp; &emsp; .blue[100 million] .font_small.gray[(Mora et al. 2011)]
### &emsp; &emsp; &emsp; living species
]]

.column[.content.center[
<br/><br/><br/>
### How many more have existed &emsp; &emsp; 
### in the past??? &emsp; 
<br/>
.font4[`r emo::ji("scream")` &emsp; ]
]]
]

???
# There are a gazillion species living, and many more so extinct
# It is a problem for big data managment and analysis
---
background-image: url("bg/5-tree-of-life-red.png")
background-position: center
background-size: contain

class: split-two

.column.split-20[
.column[.content[
]]
.column.split-20[
.row[.content[
# How do we do it?
]]
.row.split-20[
.row[.content[
## *Phylogenetic .red[trees]*
]]
.row.split-two[
.row[.content[
### - A graphical representation of a hypothesis of .red[ancestor-descendant] relationships among living and extinct species
]]
.row[.content[
### - The most important item for the study of biology and all related fields!
]]
]
]
]
]
.column[.content[
<br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/>
<br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/>
<br/><br/><br/>
<p style="font-size:0.55em"> 
<a href="https://i.redd.it/2jh0nmkx4hu11.jpg">
&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; https://i.redd.it/2jh0nmkx4hu11.jpg 
</a> </p>
]]

???

This is an artistic representation of evolutionary relationships between organisms. Although there is some information in there.
---
background-image: url("bg/6-wild.png")
background-position: center
background-size: contain

## Phylogenies in the wild

---

## A newick string

---
background-image: url("bg/7-use.png")
background-position: center
background-size: contain

# What are phylogenies useful for?


???

Conclusion: phylogenies are very important for our survival

---
background-image: url("bg/8-data-sharing.png")
background-position: center
background-size: contain

class: split-10
.column[
]
.column.split-40[
.column[.content.left[
<br/><br/><br/>
# Number of published 
# phylogenies is around 
# the .red[10K's] 
]]
.column[
]
]

???

Graphs of accumulation of phylogenies through time

---

# Hoy do you generate a phylogeny?


???

Conclusion: They are hard to make and it is a very artisanal endeavor. So you would think that scientists would make an effort to make them available.

---
background-image: url("bg/1-gray.png")
background-position: center
background-size: contain

# How do you get a published phylogeny? 
<br/><br/><br/>
## General data repositories: Supplementary Data in Journal

## Figures on paper

## Asking the authors

## Reanalyze raw data

---
background-image: url("bg/11-data-sharing2.png")
background-position: center
background-size: contain

## We might have some [data] sharing issues...

???

Main: Sharing is key for reuse of phylogenies; and reuse is key for advancement of science.

Conclusion: Phylogenies are usually not shared but DNA data is
(Drew et al. 2013).
---
background-image: url("bg/12-molecular.png")
background-position: center
background-size: contain

# Getting the raw data
## *Molecular sequence repositories*
### The International Nucleotide Sequence Database Collaboration (INSDC)

<br/>

### &emsp; &#183; .indigo[GenBank] at the National Center for Biotechnology 
### Information (NCBI) - Released .indigo[ 1989]
### https://www.ncbi.nlm.nih.gov/genbank/

<br/>

### &emsp; &#183; DNA DataBank of Japan (.orange[DDBJ]) - Released .orange[1986] 
### https://www.ddbj.nig.ac.jp/index-e.html

<br/>

### &emsp; &#183; European Nucleotide Archive (.green[ENA]) - Released .green[1982]
### https://www.ebi.ac.uk/ena  



???

More effort is going to share the raw molecular data
https://www.ddbj.nig.ac.jp/stats/release-e.html#total_data
---
background-image: url("bg/13-compare.png")
background-position: center
background-size: contain

## Public molecular data is in the order of .red[trillions]
#### Figure from https://www.ddbj.nig.ac.jp/stats/release-e.html#total_data

---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

## Let's zoom in into the GenBank database
#### &emsp; &emsp; &emsp; &emsp; Figure from https://www.nlm.nih.gov/about/2021CJ.html

<a href="https://www.nlm.nih.gov/about/2021CJ.html">
<img src="bg/14-ncbi-2019.png" width="75%" align="left" alt="ncbi logo"/>
</a>

<a href="https://www.ncbi.nlm.nih.gov/nucleotide/">
<img src="bg/12-ncbi-logo-white-bg.png" 
   style="position:fixed; right:8%; top:10%; border:none;"
   width="15%" alt="ncbi logo"/>
</a>

---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

class: split-50 center
<img src="https://i.redd.it/ez4zn3esgyw41.png" 
   style="position:fixed; left:8%; bottom:20%; border:none;"
   width="40%" alt="ncbi logo"/>
   
.column[.content.vmiddle[
# How do we go about this massive amount of raw data?
]]
.column[.content[
# The dawn of the phylogenetic .blue[pipeline] 

Phylota - Sanderson et al. 2008

Amphora - Wu et al. 2008

PHLAWD - Smith et al. 2009

PUmPER - Izquierdo et al. 2014

DarwinTree - Meng et al. 2014

NCBIminer - Xu et al. 2015

SUMAC - Freyman et al. 2015

STBase - McMahon et al. 2015

BIR - Kumar et al. 2015

SUPERSMART - Antonelli et al. 2017

SOPHI - Chesters et al. 2017

PhyloSkeleton - Guy et al. 2017

OneTwoTree - Drori et al. 2018

PhylotaR - Bennett et al. 2018

pyPHLAWD - Smith et al. 2019

Phylotol - Ceron et al. 2019
]]


???
Pipelines for using data from molecular databases

Efforts to make sense of this data: massive data pipelines

https://i.redd.it/hk7eo05svd351.jpg cats
https://i.redd.it/ez4zn3esgyw41.png brains
---
background-image: url("bg/1-red.png")
background-position: center
background-size: contain

class: split-50 center

.column[.content[
# The good

### They have contributed to fill holes on our phylogenetic knowledge of the tree of life and to maybe fix some branches.

### All these efforts contributed to start making the field a little bit more reproducible

]]
.column[.content[
# The bad

### Phylogenetic pipelines are not fully relied by community

### Manual curation always makes for better phylogenies

]]


---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

# Phylogenetic data repositories

In parallel, efforts to store actual trees and the alignments that produced them.

TreeBASE and Open Tree of Life

Conclusion: this effort is making the practice of phylogenetics more open.

---
background-image: url("bg/linux-1994.png")
background-position: center
background-size: 300px

# Phylogenetic data repositories
## TreeBASE


https://www.linuxjournal.com/content/party-its-1994
---

<div class="resp-container">
    <iframe class="testiframe" src="https://treebase.org/treebase-web/home.html" width="500px" height="600px">
    </iframe>
</div>

---
exclude: true

background-image: url("bg/treebase1.png")
background-position: center
background-size: contain

---
exclude: true

background-image: url("bg/treebase-search1.png")
background-position: center
background-size: contain

---
exclude: true

background-image: url("bg/treebase-search2.png")
background-position: center
background-size: contain

---
class: split-two

.column[.content[

<img align="center" width="550" src="bg/treebase-wget-2.gif">


````
    wget "http://purl.org/phylo/treebase/phylows/
          matrix/TB2:M4382?format=nexus"
````
]]
.column[.content[
]]


---
# Phylogenetic data repositories

## The Open Tree of Life project 

### <https://tree.opentreeoflife.org/opentree/argus/opentree12.3@ott93302>

---
background-image: url("https://github.com/OpenTreeOfLife/opentree/raw/master/webapp/static/images/logo.png")
background-position: center
background-size: 500px

class: middle, center

<div class="resp-container">
    <iframe class="testiframe" src="https://tree.opentreeoflife.org/opentree/argus/opentree12.3@ott93302" width="500px" height="600px">
    </iframe>
</div>

---

# Examples of the otol APIs and the type of data structure they return

Conclusion: we need python!

---

# Physcraper

---
class: right, middle, blue-gray

<img align="right" width="250" src="https://cdn.rawgit.com/snacktavish/physcraper/master/docs/physcraper.svg">

.font3[**Questions?**]

.font2[See you on Thursday .red[July 9], from .red[2:45]pm to .red[3:45]pm (CDT) at SciPy2020's [Biology and Bioinformatics Q&A and Moderated Discussion](https://www.scipy2020.scipy.org/schedule) live session!] 

<br/>

.font2[**Tried physcraper and have suggestions?** ]
.center[
<p style="font-size:1.55em"> Contact `r anicon::faa("envelope", animate="pulse")` us at <a href="https://mctavishlab.github.io/contact/"> ejmctavish@ucmerced.edu </a> <br/>
.red[&#183; &#183; &#183;] <br/>
Post an issue on our `r anicon::faa("github", animate="pulse")` Github repo  at <a href="https://github.com/McTavishLab/physcraper/issues">  https://github.com/McTavishLab/physcraper/issues </a> </p> 
]
---
class: split-60, blue-gray

.column[.content.vmiddle[

The University of California, Merced cluster, MERCED (Multi-Environment Research Computer for Exploration and Discovery) supported by the National Science Foundation (Grant No. ACI-1429783).

]]

.column[.content.vmiddle[
# Acknowledgements
]]

---
class: split-two, blue-gray


.column[.content.center.vmiddle.font_small[

```{r, load_refs, echo=FALSE}
library(RefManageR)
bib <- ReadBib("./assets/my_bib.bib", check = FALSE)
ui <- "- "
```

```{r, print_refs, results='asis', echo=FALSE, warning=FALSE, message=FALSE}
# writeLines(ui)
print(bib[key = "mayr2010tropical"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "mora2011how"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "drew2013lost"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "magee2014dawn"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "stenz2015exploring"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "sanmartin2007west"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "redelings2017supertree"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "magallon2019thirty"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "sakamoto2016dinosaurs"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "gaunt2001phylogenies"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
print(bib[key = "de2018recent"], 
  .opts = list(check.entries = FALSE, 
               style = "html", 
               bib.style = "authoryear"))
```

]]
.column[.content.vmiddle.center[
# References
]]

???

code from https://github.com/yihui/xaringan/issues/26
