<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>SciPy 2020</title>
    <meta charset="utf-8" />
    <link href="libs/remark-css/kunoichi.css" rel="stylesheet" />
    <link href="libs/remark-css/ninjutsu.css" rel="stylesheet" />
    <link href="libs/remark-css/metropolis-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome-animation/font-awesome-animation-emi.css" rel="stylesheet" />
    <script src="libs/fontawesome/js/fontawesome-all.min.js"></script>
    <link rel="stylesheet" href="ninjutsu-luna-extras.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">

background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

class: left, middle, blue-gray

&lt;img align="left" width="250" src="https://cdn.rawgit.com/snacktavish/physcraper/master/docs/physcraper.svg"&gt;




.font3[**Physcraper**]

.font2[Continual .red[updating] of evolutionary estimates in the Open Tree of Life]

&lt;br/&gt;
### .red[&lt;sup&gt;*&lt;/sup&gt;] Luna L. Sanchez Reyes .red[&amp;#183;]  Martha Kandziora .red[&amp;#183;] Emily Jane McTavish
&lt;br/&gt;
### University of California, Merced
&lt;br/&gt;


SciPy2020 Conference .red[&amp;#183;] July 6-12, 2020 .red[&amp;#183;] The Internet

Slides last updated on  June 15, 2020 .red[&amp;#183;] Find the code for them in [<span>&lt;i class="fab  fa-github faa-passing animated "&gt;&lt;/i&gt;</span>  here]()

.red[*] Speaker contact [@LunaSare](https://www.lunasare.com/) on [Github <span>&lt;i class="fab  fa-github faa-float animated "&gt;&lt;/i&gt;</span>](https://github.com/LunaSare) and [Twitter <span>&lt;i class="fab  fa-twitter faa-float animated "&gt;&lt;/i&gt;</span>](https://twitter.com/LunaSare)
---

background-image: url("bg/2-past-future1.png")
background-position: center
background-size: contain

???

# To understand our future you need to understand where you come from.

---
background-image: url("bg/3-diversity1.png")
background-position: center
background-size: contain

???

# As biologists, we want to understand how all living things are connected evolutionarily

---
background-image: url("bg/4-diversity2.png")
background-position: center
background-size: contain

class: split-20, center

.row[.content[
# Let's talk scale... .orange[how many] organisms are there?

]]
.row.split-two[
.column[.content.center[
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
### &amp;emsp; &amp;emsp; &amp;emsp;  Estimates range between
## &amp;emsp; &amp;emsp; &amp;emsp; .orange[3 million] .font_small.gray[(Mayr 2010)]
### &amp;emsp; &amp;emsp; &amp;emsp;  and
## &amp;emsp; &amp;emsp; &amp;emsp; .blue[100 million] .font_small.gray[(Mora et al. 2011)]
### &amp;emsp; &amp;emsp; &amp;emsp; living species
]]

.column[.content.center[
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
### How many more have existed &amp;emsp; &amp;emsp;
### in the past??? &amp;emsp;
&lt;br/&gt;
.font4[😱 &amp;emsp; ]
]]
]

???
# There are a gazillion species living, and many more so extinct
# It is a problem for big data managment and analysis
---
background-image: url("bg/5-tree-of-life-red.png")
background-position: center
background-size: contain

class: split-two

.column.split-20[
.column[.content[
]]
.column.split-20[
.row[.content[
# How can we understand the past?
]]
.row.split-20[
.row[.content[
## *Phylogenetic .red[trees]*
]]
.row.split-two[
.row[.content[
### - A graphical representation of a hypothesis of .red[ancestor-descendant] relationships among living and extinct species
]]
.row[.content[
### - The most important item for the study of biology and all related fields!
]]
]
]
]
]
.column[.content[
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
&lt;p style="font-size:0.55em"&gt;
&lt;a href="https://i.redd.it/2jh0nmkx4hu11.jpg"&gt;
&amp;emsp; &amp;emsp; &amp;emsp; &amp;emsp; &amp;emsp; &amp;emsp; https://i.redd.it/2jh0nmkx4hu11.jpg
&lt;/a&gt; &lt;/p&gt;
]]

???

This is an artistic representation of evolutionary relationships between organisms. Although there is some information in there.
---
background-image: url("bg/6-wild.png")
background-position: center
background-size: contain

# Phylogenies in the .orange[wild]

---
class: split-two

.column[.content.center[
# What .blue[we] see...
&lt;br/&gt;&lt;br/&gt;
&lt;img src="bg/6-de2018recent-fig4c.png" width="100%" align="center" alt="Fouquieria Diversification rates"/&gt;

.font_small.gray[Figure 4c from De Nova et al. 2018]

&lt;br/&gt;&lt;br/&gt;
.font2.blue[A pdf file]

]]
.column[.content.center[
# What a .lime[computer] sees...

&lt;p style="font-size:0.70em"&gt;
TREE * STATE_6680000 = [&amp;R] ((2:23.9912476,1:23.9912476):67.64363987,(((((((((13:0.4486448057,18:0.4486448057):0.419000619,12:0.8676454246):0.1015486898,11:0.9691941144):0.03970804765,15:1.008901162):0.4621950546,(21:0.5368201226,14:0.5368191226):0.9342770941):1.047882735,((19:0.9651292965,(16:0.9169158264,17:0.9169148264):0.04821447015):1.162235908,20:2.127366205):0.3916137472):3.896603157,(9:3.020538176,10:3.020538176):3.395044933):10.55399955,(((((((((55:0.8502407373,58:0.8502407373):0.3679373142,(53:0.4864845726,56:0.4864845726):0.731693479):0.3563793329,54:1.574557384):0.09035613592,57:1.66491352):1.88446994,(49:0.8161479593,45:0.8161479593):2.733234501):0.4760065168,(46:2.107086476,(47:0.9224473608,48:0.9224473608):1.184639116):1.918302501):1.583903154,((3:1.071332259,28:1.071332259):3.013986475,(23:4.030250514,((((24:0.765467551,22:0.765466551):0.9079596426,25:1.673426194):1.323739158,((4:1.813102509,(6:0.6164327737,(7:0.3945867261,8:0.3945867261):0.2218450476):1.196669735):0.628500496,(26:1.575611873,5:1.575611873):0.8659911318):0.5555633466):0.282086233,27:3.279251584):0.7509989298):0.05506722029):1.523974397):1.046228415,((((78:1.001185246,79:1.001185246):1.6203738,(77:1.912610336,(80:1.126503072,81:1.126503072):0.7861072638):0.7089487102):0.9192230645,(74:1.580125802,((73:0.9320116249,76:0.9320116249):0.4249115549,75:1.35692418):0.2232026218):1.960657309):1.751839439,(51:1.304268197,(52:0.9244796054,50:0.9244796054):0.3797885921):3.988354352):1.362898998):3.327530237,(43:4.486855597,44:4.486855597):5.496195188):6.986530871):1.541020887,(((((42:0.3927064654,40:0.3927064654):1.216825565,29:1.609532031):0.3505658459,((33:0.05906297464,41:0.05906297464):1.012591397,30:1.071654371):0.8884435051):0.8794856624,(31:0.7769783382,(39:0.6041099539,32:0.6041099539):0.1728683843):2.062605201):4.0079666,(((35:3.024986853,38:3.024986853):0.3694745008,(36:0.7769265764,(37:0.60170545,34:0.60170545):0.1752211264):2.617534778):2.437537487,(71:3.496481764,(((69:0.6822085102,68:0.6822085102):0.2919955037,70:0.9742040138):2.491583884,(((64:2.639693826,(60:1.924607493,67:1.924607493):0.7150863324):0.1614034831,((63:1.369885952,66:1.369885952):0.7113535387,(61:0.7383880773,65:0.7383880773):1.342851413):0.7198578182):0.4360311418,((72:1.070412672,59:1.070412672):1.754535535,62:2.824948207):0.4121802435):0.2286584466):0.03069486699):2.335517077):1.015550298):11.6630534):73.12428394);
&lt;/p&gt;

.font2.lime[A newick string]

]]

---
background-image: url("bg/7-use2.png")
background-position: center
background-size: contain

# What are .gray[phylogenies useful] for?


???

Conclusion: phylogenies are very important for our survival

---
background-image: url("bg/8-data-sharing.png")
background-position: center
background-size: contain

class: split-10
.column[
]
.column.split-40[
.column[.content.left[
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
# Number of published
# phylogenies is around
# the .red[10K's]
]]
.column[
]
]

???

Graphs of accumulation of phylogenies through time

---

class: split-10 center

.row.split-1-2-1.bg-gray[
.column[

]
.column.bg-white[.content.vmiddle.center[
# How do we .orange[generate] a phylogeny?
]]
.column[

]
]
.row.split-five[
.column.split-20[
.row[.content.vmiddle[
(0) Get some money.
]]
.row[.content[
&lt;img src="bg/10-money-virus-4970581_1920.jpg" width="100%" alt = "money from academia"/&gt;
&lt;img src="bg/10-money-arm-wrestling-567950_1920.jpg" width="100%" alt = "money from other sources"/&gt;
&lt;img src="bg/10-money-grow-2724241_1920.jpg" width="100%" alt = "money from your investments"/&gt;

]]
]
.column.split-30[
.row[.content.vmiddle[
(1) Get some raw data: catch some organisms, or get them from a museum; get some characters to analyze, preferably DNA.
]]
.row[.content[
&lt;img src="bg/10-dna-mgid_arc_content_nickelodeon.com.jpeg" width="100%" alt = "Spongebob catching jellyfish"/&gt;
&lt;img src="bg/10-adn-modified-1744952_1920.jpg" width="100%" alt = "extract DNA"/&gt;
]]
]
.column.split-40[
.row[.content.vmiddle[
(2) Assemble a hypothesis of homology. If you choose molecular characters, align the sequences!

]]
.row[.content[
&lt;img src="bg/10-align-laptop-3087585_1920.jpg" width="100%" alt = ""/&gt;
&lt;img src="https://upload.wikimedia.org/wikipedia/commons/7/79/RPLP0_90_ClustalW_aln.gif" width="100%" alt = "wikipedia protein alignment"/&gt;

]]
]
.column.split-30[
.row[.content.vmiddle[
(3) Analyse this hypothesis of homology to create a hypothesis of phylogenetic relationships. There are different philosophies!
]]
.row[.content[
&lt;img src="bg/10-tree-budha-sri-lanka-1024347_1920.jpg" width="100%" alt = "budhism"/&gt;
&lt;img src="bg/10-tree-bustos-756620_1920.jpg" width="100%" alt = "greek philosophers"/&gt;

]]

]
.column.split-20[
.row[.content.vmiddle[
(4) Compare and discuss the inferred relationships in the context of previous hypothesis.
]]
.row[.content[
&lt;img src="bg/10-discuss-giraffes-627031_1920.jpg" width="100%" alt = "giraffes discussing"/&gt;
&lt;img src="bg/10-discuss-sheep-3557445_1920.jpg" width="100%" alt = "moutons discussing"/&gt;
&lt;img src="bg/10-discuss-sparrows-2759978_1920.jpg" width="100%" alt = "sparrows discussing"/&gt;
]]
]
]

???

Conclusion: They are hard to make and it is a very artisanal endeavor. So you would think that scientists would make an effort to make them available.

&lt;img src="bg/10-dna-shine-2649850_1920.jpg" width="100%" alt = "sequence the DNA"/&gt;
extract their DNA, and sequence it, or skip all that and get the DNA from an online nucleotide data repository. There are a bunch and we will get to that in a minute.

phylogenetic relationships among the organisms you collected.
---

class: split-10 center

.row.split-60.bg-gray[
.column.bg-white[.content.vmiddle.center[
# How can we get a .orange[published] phylogeny?
]]
.column[
]
]
.row.split-two[
.row.split-four[
.column[.content.vmiddle[
## General data repositories: Supplementary Data in Journal
]]
.column[.content.vmiddle[
&lt;img src="bg/11-suppl-files-1614223_1920.jpg" width="100%" align="center" alt="a paper archive"/&gt;

]]
.column[.content.vmiddle[
&lt;img src="bg/6-de2018recent-fig4c.png" width="100%" align="center" alt="Fouquieria diversification rates"/&gt;

.font_small.gray[Figure 4c from De Nova et al. 2018]

.blue[A pdf file]
]]
.column[.content.vmiddle[
## Figures on paper
]]
]
.row.split-two[
.column[.content.vmiddle[
## Asking the authors
&lt;img src="http://www.phdcomics.com/comics/archive/phd072508s.gif" width="100%" alt = "wikipedia protein alignment"/&gt;
]]
.column[.content.vmiddle[
## Reanalyzing the raw data
&lt;img src="https://upload.wikimedia.org/wikipedia/commons/7/79/RPLP0_90_ClustalW_aln.gif" width="60%" alt = "wikipedia protein alignment"/&gt;
]]
]
]

---
background-image: url("bg/11-data-sharing2.png")
background-position: center
background-size: contain

## There might be some *data* sharing issues...

???

Main: Sharing is key for reuse of phylogenies; and reuse is key for advancement of science.

Conclusion: Phylogenies are usually not shared but DNA data is
(Drew et al. 2013).
---
background-image: url("bg/12-molecular.png")
background-position: center
background-size: contain

# Getting the raw data
## *Molecular sequence repositories*
### The International Nucleotide Sequence Database Collaboration (INSDC)

&lt;br/&gt;

### &amp;emsp; &amp;#183; .indigo[GenBank] at the National Center for Biotechnology
### Information (NCBI) - Released .indigo[ 1989]
### https://www.ncbi.nlm.nih.gov/genbank/

&lt;br/&gt;

### &amp;emsp; &amp;#183; DNA DataBank of Japan (.orange[DDBJ]) - Released .orange[1986]
### https://www.ddbj.nig.ac.jp/index-e.html

&lt;br/&gt;

### &amp;emsp; &amp;#183; European Nucleotide Archive (.green[ENA]) - Released .green[1982]
### https://www.ebi.ac.uk/ena



???

More effort is going to share the raw molecular data
https://www.ddbj.nig.ac.jp/stats/release-e.html#total_data
---
background-image: url("bg/13-compare.png")
background-position: center
background-size: contain

## Public molecular data is in the order of .red[trillions]
#### Figure from https://www.ddbj.nig.ac.jp/stats/release-e.html#total_data

---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

## Let's zoom in into the GenBank database
#### &amp;emsp; &amp;emsp; &amp;emsp; &amp;emsp; Figure from https://www.nlm.nih.gov/about/2021CJ.html

&lt;a href="https://www.nlm.nih.gov/about/2021CJ.html"&gt;
&lt;img src="bg/14-ncbi-2019.png" width="75%" align="left" alt="ncbi logo"/&gt;
&lt;/a&gt;

&lt;a href="https://www.ncbi.nlm.nih.gov/nucleotide/"&gt;
&lt;img src="bg/12-ncbi-logo-white-bg.png"
   style="position:fixed; right:8%; top:10%; border:none;"
   width="15%" alt="ncbi logo"/&gt;
&lt;/a&gt;

---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

class: center

# How do we go about this .blue[massive] amount of raw .red[data]?

---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

class: split-70 center

.column.split-20[
.row[.content.vmiddle[
# The dawn of the phylogenetic .blue[pipeline]
]]
.row.split-two[
.column[.content.vmiddle[
Phylota - Sanderson et al. 2008

Amphora - Wu et al. 2008

PHLAWD - Smith et al. 2009

PUmPER - Izquierdo et al. 2014

DarwinTree - Meng et al. 2014

NCBIminer - Xu et al. 2015

SUMAC - Freyman et al. 2015

STBase - McMahon et al. 2015
]]
.column[.content.vmiddle[
BIR - Kumar et al. 2015

SUPERSMART - Antonelli et al. 2017

SOPHI - Chesters et al. 2017

PhyloSkeleton - Guy et al. 2017

OneTwoTree - Drori et al. 2018

PhylotaR - Bennett et al. 2018

pyPHLAWD - Smith et al. 2019

Phylotol - Ceron et al. 2019
]]
]
]
.column.bg-blue[.content.vmiddle[
&lt;img src="https://i.redd.it/ez4zn3esgyw41.png"
   width="100%" alt="pipeline meme"/&gt;
]]

???
Pipelines for using data from molecular databases

Efforts to make sense of this data: massive data pipelines

https://i.redd.it/hk7eo05svd351.jpg cats
https://i.redd.it/ez4zn3esgyw41.png brains
---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

class: split-10 center

.column.bg-blue[

]
.column.split-20[
.row[.content.vmiddle[
# The phylogenetic .blue[pipeline]
]]
.row.split-two[
.column[.content.vmiddle[
# The good 👍
&lt;br/&gt;
### They have contributed to fill holes on our phylogenetic knowledge of the tree of life and to maybe fix some branches.
&lt;br/&gt;
### All these efforts contributed to start making the field a little bit more reproducible

]]
.column[.content.vmiddle[
# The bad 👎
&lt;br/&gt;
### Phylogenetic pipelines are not fully relied by community
&lt;br/&gt;
### Manual curation always makes for better phylogenies

]]
]
]
---
background-image: url("bg/1-blue.png")
background-position: center
background-size: contain

class: split-30 center

.row[.content.vmiddle.center[
# Getting the tree in computer readable format
## *Phylogenetic tree data repositories*
]]

.row.split-two[
.row.split-70[
.column[.content[
### &amp;emsp; &amp;#183; TreeBASE - Released .indigo[1994]
### https://treebase.org/treebase-web/home.html
### If the TreeBASE webpage is lagging, go [here]() for a demo.
]]
.column[.content[
&lt;img src="bg/linux-1994.png" width="75%" align="left" alt="treebase logo"/&gt;
.font_small[https://www.linuxjournal.com/content/party-its-1994]
]]
]
.row.split-70[
.column[.content[

### &amp;emsp; &amp;#183; The  .orange[Open Treeof Life] - Released .orange[2010]
### &lt;https://tree.opentreeoflife.org/opentree/argus/opentree12.3@ott93302&gt;
]]
.column[.content[
&lt;img src="" width="75%" align="left" alt="opentree logo"/&gt;

]]
]
]




???

# Phylogenetic data repositories

Efforts to store actual trees and the alignments that produced them.
These efforts happened in parallel to the development of pipelines.

TreeBASE and Open Tree of Life

Conclusion: this effort is making the practice of phylogenetics more open.

---
exclude: true

background-image: url("bg/treebase1.png")
background-position: center
background-size: contain

---
exclude: true

background-image: url("bg/treebase-search1.png")
background-position: center
background-size: contain

---
exclude: true

background-image: url("bg/treebase-search2.png")
background-position: center
background-size: contain

---
class: split-two

.column[.content[

&lt;img align="center" width="550" src="bg/treebase-wget-2.gif"&gt;


````
    wget "http://purl.org/phylo/treebase/phylows/
          matrix/TB2:M4382?format=nexus"
````
]]
.column[.content[
]]

---

# Examples of the otol APIs and the type of data structure they return

Conclusion: we need python!

---

# Physcraper

---
class: right, middle, blue-gray

&lt;img align="right" width="250" src="https://cdn.rawgit.com/snacktavish/physcraper/master/docs/physcraper.svg"&gt;

.font3[**Questions?**]

.font2[See you on Thursday .red[July 9], from .red[2:45]pm to .red[3:45]pm (CDT) at SciPy2020's [Biology and Bioinformatics Q&amp;A and Moderated Discussion](https://www.scipy2020.scipy.org/schedule) live session!]

&lt;br/&gt;

.font2[**Tried physcraper and have suggestions?** ]
.center[
&lt;p style="font-size:1.55em"&gt; Contact <span>&lt;i class="fas  fa-envelope faa-pulse animated "&gt;&lt;/i&gt;</span> us at &lt;a href="https://mctavishlab.github.io/contact/"&gt; ejmctavish@ucmerced.edu &lt;/a&gt; &lt;br/&gt;
.red[&amp;#183; &amp;#183; &amp;#183;] &lt;br/&gt;
Post an issue on our <span>&lt;i class="fab  fa-github faa-pulse animated "&gt;&lt;/i&gt;</span> Github repo  at &lt;a href="https://github.com/McTavishLab/physcraper/issues"&gt;  https://github.com/McTavishLab/physcraper/issues &lt;/a&gt; &lt;/p&gt;
]
---
class: split-60, blue-gray

.column[.content.vmiddle[

The University of California, Merced cluster, MERCED (Multi-Environment Research Computer for Exploration and Discovery) supported by the National Science Foundation (Grant No. ACI-1429783).

]]

.column[.content.vmiddle[
# Acknowledgements
]]

---
class: split-two, blue-gray


.column[.content.center.vmiddle.font_small[



&lt;p&gt;&lt;cite&gt;Mayr, R. M.
(2010).
&amp;ldquo;Tropical Arthropod Species, More or Less?&amp;rdquo;.
In: &lt;em&gt;Science&lt;/em&gt; 329.July, pp. 41&amp;ndash;42.
URL: &lt;a href="http://science.sciencemag.org/content/329/5987/41/tab-pdf"&gt;http://science.sciencemag.org/content/329/5987/41/tab-pdf&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Mora, C., D. P. Tittensor, S. Adl, et al.
(2011).
&amp;ldquo;How many species are there on earth and in the ocean?&amp;rdquo;.
In: &lt;em&gt;PLoS Biology&lt;/em&gt; 9.8, pp. 1&amp;ndash;8.
ISSN: 15449173.
DOI: &lt;a href="https://doi.org/10.1371/journal.pbio.1001127"&gt;10.1371/journal.pbio.1001127&lt;/a&gt;.
eprint: NIHMS150003.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Drew, B. T., R. Gazis, P. Cabezas, et al.
(2013).
&amp;ldquo;Lost branches on the tree of life&amp;rdquo;.
In: &lt;em&gt;PLoS biology&lt;/em&gt; 11.9.
DOI: &lt;a href="https://doi.org/10.1371/journal.pbio.1001636"&gt;10.1371/journal.pbio.1001636&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Magee, A. F., M. R. May, and B. R. Moore
(2014).
&amp;ldquo;The dawn of open access to phylogenetic data&amp;rdquo;.
In: &lt;em&gt;PLoS One&lt;/em&gt; 9.10.
DOI: &lt;a href="https://doi.org/10.1371/journal.pone.0110268"&gt;10.1371/journal.pone.0110268&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Stenz, N. W., B. Larget, D. A. Baum, et al.
(2015).
&amp;ldquo;Exploring tree-like and non-tree-like patterns using genome sequences: an example using the inbreeding plant species Arabidopsis thaliana (L.) Heynh&amp;rdquo;.
In: &lt;em&gt;Systematic biology&lt;/em&gt; 64.5, pp. 809&amp;ndash;823.
DOI: &lt;a href="https://doi.org/10.1093/sysbio/syv039"&gt;10.1093/sysbio/syv039&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Sanmartín, I., L. Wanntorp, and R. C. Winkworth
(2007).
&amp;ldquo;West Wind Drift revisited: testing for directional dispersal in the Southern Hemisphere using event-based tree fitting&amp;rdquo;.
In: &lt;em&gt;Journal of Biogeography&lt;/em&gt; 34.3, pp. 398&amp;ndash;416.
DOI: &lt;a href="https://doi.org/10.1111/j.1365-2699.2006.01655.x"&gt;10.1111/j.1365-2699.2006.01655.x&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Redelings, B. D. and M. T. Holder
(2017).
&amp;ldquo;A supertree pipeline for summarizing phylogenetic and taxonomic information for millions of species&amp;rdquo;.
In: &lt;em&gt;PeerJ&lt;/em&gt; 5, p. e3058.
DOI: &lt;a href="https://doi.org/10.7717/peerj.3058"&gt;10.7717/peerj.3058&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Magallón, S., L. L. Sánchez-Reyes, and S. L. Gómez-Acevedo
(2019).
&amp;ldquo;Thirty clues to the exceptional diversification of flowering plants&amp;rdquo;.
In: &lt;em&gt;Annals of botany&lt;/em&gt; 123.3, pp. 491&amp;ndash;503.
DOI: &lt;a href="https://doi.org/10.1093/aob/mcy182"&gt;10.1093/aob/mcy182&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Sakamoto, M., M. J. Benton, and C. Venditti
(2016).
&amp;ldquo;Dinosaurs in decline tens of millions of years before their final extinction&amp;rdquo;.
In: &lt;em&gt;Proceedings of the National Academy of Sciences&lt;/em&gt; 113.18, pp. 5036&amp;ndash;5040.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Gaunt, M., S. Turner, L. Rigottier-Gois, et al.
(2001).
&amp;ldquo;Phylogenies of atpD and recA support the small subunit rRNA-based classification of rhizobia.&amp;rdquo;
In: &lt;em&gt;International journal of systematic and evolutionary microbiology&lt;/em&gt; 51.6, pp. 2037&amp;ndash;2048.
DOI: &lt;a href="https://doi.org/10.1099/00207713-51-6-2037"&gt;10.1099/00207713-51-6-2037&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;De-Nova, J. A., L. L. Sánchez-Reyes, L. E. Eguiarte, et al.
(2018).
&amp;ldquo;Recent radiation and dispersal of an ancient lineage: The case of Fouquieria (Fouquiericeae, Ericales) in North American deserts&amp;rdquo;.
In: &lt;em&gt;Molecular phylogenetics and evolution&lt;/em&gt; 126, pp. 92&amp;ndash;104.
DOI: &lt;a href="https://doi.org/10.1016/j.ympev.2018.03.026"&gt;10.1016/j.ympev.2018.03.026&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;

]]
.column[.content.vmiddle.center[
# References
]]

???

code from https://github.com/yihui/xaringan/issues/26
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9",
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
