<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <title>SciPy 2020</title>
    <meta charset="utf-8" />
    <link href="libs/remark-css/kunoichi.css" rel="stylesheet" />
    <link href="libs/remark-css/ninjutsu.css" rel="stylesheet" />
    <link href="libs/remark-css/metropolis-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome-animation/font-awesome-animation-emi.css" rel="stylesheet" />
    <script src="libs/fontawesome/js/fontawesome-all.min.js"></script>
    <link rel="stylesheet" href="ninjutsu-luna-extras.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">

background-image: url("bg/2-blue-sky.png")
background-position: center
background-size: contain

class: left, middle, blue-gray

&lt;img align="left" width="250" src="https://cdn.rawgit.com/snacktavish/physcraper/master/docs/physcraper.svg"&gt;




.font3[**Physcraper**]

.font2[Continual .red[updating] of evolutionary estimates in the Open Tree of Life]

&lt;br/&gt;
### .red[&lt;sup&gt;*&lt;/sup&gt;] Luna L. Sanchez Reyes .red[&amp;#183;]  Martha Kandziora .red[&amp;#183;] Emily Jane McTavish
&lt;br/&gt;
### University of California, Merced
&lt;br/&gt;


SciPy2020 Conference .red[&amp;#183;] July 6-12, 2020 .red[&amp;#183;] The Internet

Slides last updated on  June 11, 2020 .red[&amp;#183;] Find the code for them in [<span>&lt;i class="fab  fa-github faa-passing animated "&gt;&lt;/i&gt;</span>  here]()

.red[*] Speaker contact [@LunaSare](https://www.lunasare.com/) on [Github <span>&lt;i class="fab  fa-github faa-float animated "&gt;&lt;/i&gt;</span>](https://github.com/LunaSare) and [Twitter <span>&lt;i class="fab  fa-twitter faa-float animated "&gt;&lt;/i&gt;</span>](https://twitter.com/LunaSare)
---

background-image: url("bg/past-future1.png")
background-position: center
background-size: contain

???

# To understand our future you need to understand where you come from.

---
background-image: url("bg/diversity1.png")
background-position: center
background-size: contain

???

# As biologists, we want to understand how all living things are connected evolutionarily

---
background-image: url("bg/diversity2.png")
background-position: center
background-size: contain

class: split-20, center

.row[.content[
# Let's talk scale... how many organisms are there?

]]
.row.split-two[
.column[.content.center[
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
### &amp;emsp; &amp;emsp; &amp;emsp;  Estimates range between 
## &amp;emsp; &amp;emsp; &amp;emsp; .orange[3 million] .font_small.gray[(Mayr 2010)]
### &amp;emsp; &amp;emsp; &amp;emsp;  and 
## &amp;emsp; &amp;emsp; &amp;emsp; .blue[100 million] .font_small.gray[(Mora et al. 2011)]
### &amp;emsp; &amp;emsp; &amp;emsp; living species
]]

.column[.content.center[
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
### How many more have existed &amp;emsp; &amp;emsp; 
### in the past??? &amp;emsp; 
&lt;br/&gt;
.font4[ðŸ˜± &amp;emsp; ]
]]
]

???
# There are a gazillion species living, and many more so extinct
# It is a problem for big data managment and analysis
---
background-image: url("bg/tree-of-life-red.png")
background-position: center
background-size: contain

class: split-two

.column.split-20[
.column[.content[
]]
.column.split-20[
.row[.content[
# How do we do it?
]]
.row.split-20[
.row[.content[
## Phylogenetic .red[trees]
]]
.row.split-two[
.row[.content[
### - A graphical representation of a hypothesis of .red[ancestor-descendant] relationships among living and extinct species
]]
.row[.content[
### - The most important item for the study of biology and all related fields!
]]
]
]
]
]
.column[.content[
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
&lt;p style="font-size:0.55em"&gt; 
&lt;a href="https://i.redd.it/2jh0nmkx4hu11.jpg"&gt;
&amp;emsp; &amp;emsp; &amp;emsp; &amp;emsp; &amp;emsp; &amp;emsp; https://i.redd.it/2jh0nmkx4hu11.jpg 
&lt;/a&gt; &lt;/p&gt;
]]

???

This is an artistic representation of evolutionary relationships between organisms. Although there is some information in there.
---

## Phylogenies in the wild

---
class: split-20
# Use of phylogenies in science


Images of modern phylogenies

???

Conclusion: phylogenies are very important for our survival

---

# Hoy do you produce them?


???

Conclusion: They are hard to make and it is a very artisanal endeavor. So you would think that scientists would make an effort to make them available.

---
background-image: url("bg/data-sharing1.png")
background-position: center
background-size: contain

# Because they are important, there are a lot of phylogenies

???

Graphs of accumulation of phylogenies through time

---
background-image: url("bg/2-gray.png")
background-position: center
background-size: contain

# How do you get them? Part1

Data repositories: Supplementary Data in Journal

Figures

Original files

---
background-image: url("bg/data-sharing2.png")
background-position: center
background-size: contain

# How do you get them? Part2

Main: Sharing is key for reuse of phylogenies; and reuse is key for advancement of science.

???

Conclusion: Phylogenies are usually not shared but DNA data is
(Drew et al. 2013).
---

# Molecular data repositories


NCBI - 1989

EMBL - 

???

More effort is goinf to share the raw molecular data

---

&lt;div class="resp-container"&gt;
    &lt;iframe class="testiframe" src="https://www.ncbi.nlm.nih.gov/nucleotide/" height="600px"&gt;
    &lt;/iframe&gt;
&lt;/div&gt;

---

# Pipelines for using data from molecular databases

Efforts to make sense of this data: massive data pipelines


Phylota - 2008


---

# Phylogenetic pipelines not fully relied by community

Main idea: manual curation always makes better phylogenies

Conclusion: All these efforts contributed to start making the field a little bit more reproducible

---

# Phylogenetic data repositories

In parallel, efforts to store actual trees and the alignments that produced them.

TreeBASE and Open Tree of Life

Conclusion: this effort is making the practice of phylogenetics more open.

---
background-image: url("bg/linux-1994.png")
background-position: center
background-size: 300px

# Phylogenetic data repositories

## TreeBASE


https://www.linuxjournal.com/content/party-its-1994
---

&lt;div class="resp-container"&gt;
    &lt;iframe class="testiframe" src="https://treebase.org/treebase-web/home.html" width="500px" height="600px"&gt;
    &lt;/iframe&gt;
&lt;/div&gt;

---

background-image: url("bg/treebase1.png")
background-position: center
background-size: contain

---

background-image: url("bg/treebase-search1.png")
background-position: center
background-size: contain

---

background-image: url("bg/treebase-search2.png")
background-position: center
background-size: contain

---
class: split-two

.column[.content[
```
wget "http://purl.org/phylo/treebase/phylows/matrix/TB2:M4382?format=nexus"
```
]]
.column[.content[
&lt;img align="right" width="550" src="bg/treebase-wget-2.gif"&gt;
]]


---
# Phylogenetic data repositories

## The Open Tree of Life project 

### &lt;https://tree.opentreeoflife.org/opentree/argus/opentree12.3@ott93302&gt;

---
background-image: url("https://github.com/OpenTreeOfLife/opentree/raw/master/webapp/static/images/logo.png")
background-position: center
background-size: 500px

class: middle, center

&lt;div class="resp-container"&gt;
    &lt;iframe class="testiframe" src="https://tree.opentreeoflife.org/opentree/argus/opentree12.3@ott93302" width="500px" height="600px"&gt;
    &lt;/iframe&gt;
&lt;/div&gt;

---

# Examples of the otol APIs and the type of data structure they return

Conclusion: we need python!

---

# Physcraper

---
class: right, middle, blue-gray

&lt;img align="right" width="250" src="https://cdn.rawgit.com/snacktavish/physcraper/master/docs/physcraper.svg"&gt;

.font3[**Questions?**]

.font2[See you on Thursday .red[July 9], from .red[2:45]pm to .red[3:45]pm (CDT) at SciPy2020's [Biology and Bioinformatics Q&amp;A and Moderated Discussion](https://www.scipy2020.scipy.org/schedule) live session!] 

&lt;br/&gt;

.font2[**Tried physcraper and have suggestions?** ]
.center[
&lt;p style="font-size:1.55em"&gt; Contact <span>&lt;i class="fas  fa-envelope faa-pulse animated "&gt;&lt;/i&gt;</span> us at &lt;a href="https://mctavishlab.github.io/contact/"&gt; ejmctavish@ucmerced.edu &lt;/a&gt; &lt;br/&gt;
.red[&amp;#183; &amp;#183; &amp;#183;] &lt;br/&gt;
Post an issue on our <span>&lt;i class="fab  fa-github faa-pulse animated "&gt;&lt;/i&gt;</span> Github repo  at &lt;a href="https://github.com/McTavishLab/physcraper/issues"&gt;  https://github.com/McTavishLab/physcraper/issues &lt;/a&gt; &lt;/p&gt; 
]
---
class: split-80, blue-gray


.column[.content.center.vmiddle[



&lt;p&gt;&lt;cite&gt;Mayr, R. M.
(2010).
&amp;ldquo;Tropical Arthropod Species, More or Less?&amp;rdquo;.
In: &lt;em&gt;Science&lt;/em&gt; 329.July, pp. 41&amp;ndash;42.
URL: &lt;a href="http://science.sciencemag.org/content/329/5987/41/tab-pdf"&gt;http://science.sciencemag.org/content/329/5987/41/tab-pdf&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Mora, C., D. P. Tittensor, S. Adl, et al.
(2011).
&amp;ldquo;How many species are there on earth and in the ocean?&amp;rdquo;.
In: &lt;em&gt;PLoS Biology&lt;/em&gt; 9.8, pp. 1&amp;ndash;8.
ISSN: 15449173.
DOI: &lt;a href="https://doi.org/10.1371/journal.pbio.1001127"&gt;10.1371/journal.pbio.1001127&lt;/a&gt;.
eprint: NIHMS150003.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Drew, B. T., R. Gazis, P. Cabezas, et al.
(2013).
&amp;ldquo;Lost branches on the tree of life&amp;rdquo;.
In: &lt;em&gt;PLoS biology&lt;/em&gt; 11.9.
DOI: &lt;a href="https://doi.org/10.1371/journal.pbio.1001636"&gt;10.1371/journal.pbio.1001636&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;
&lt;p&gt;&lt;cite&gt;Magee, A. F., M. R. May, and B. R. Moore
(2014).
&amp;ldquo;The dawn of open access to phylogenetic data&amp;rdquo;.
In: &lt;em&gt;PLoS One&lt;/em&gt; 9.10.
DOI: &lt;a href="https://doi.org/10.1371/journal.pone.0110268"&gt;10.1371/journal.pone.0110268&lt;/a&gt;.&lt;/cite&gt;&lt;/p&gt;

]]
.column[.content.vmiddle.center[
# References
]]

???

code from https://github.com/yihui/xaringan/issues/26
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"ratio": "16:9",
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
